common/src/main/java/com/redhat/lightblue/rest/logging/LoggingFilter.java

```diff
@@ -0,0 +1,65 @@
+package com.redhat.lightblue.rest.logging;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+@WebFilter(urlPatterns = {"/*"})
+public class LoggingFilter implements Filter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingFilter.class);
+
+    public static final String HEADER_REQUEST_UUID = "RequestUUID";
+
+    @Override
+    public void destroy() {
+        //Do Nothing!
+    }
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain) throws IOException, ServletException {
+        try {
+            String requestUUID = UUID.randomUUID().toString();
```

Comment:
The idea is to have a unique identifier for all logs produced within a session for all time. For example, if I knew the uuid for an exception I could grep all the logs for that uuid for all logs associated with the exception and hopefully get a more complete picture of what happened.

From what I can tell, your suggestion would achieve that goal (although we might want to make it volatile also) so long as the application isn't restarted. At which point the Integer starts over. So when grep is performed on the logs now, you could get multiple connection log sets. Is there a way to make AtomicInteger unique across restarts? I suppose we could persist the value to disk, but that seems like a lot of work compared to just using a uuid.